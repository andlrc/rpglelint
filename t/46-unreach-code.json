[
   {
      "column":3,
      "file":"t/46-unreach-code.rpgle",
      "line":10,
      "message":"code will never be executed",
      "severity":"warning"
   }
]
