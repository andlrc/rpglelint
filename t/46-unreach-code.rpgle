// rpglelint: -Wunreachable-code
dcl-proc proc;

  if a > 1;
    exsr subr1;
  endif;

  exsr subr2;
  return 'don''t fail here';
  return 'fail here';

  begsr subr1;
    return 'return in if-statement';
  endsr;

  begsr subr2;
  endsr;
end-proc;
