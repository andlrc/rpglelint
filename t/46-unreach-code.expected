t/46-unreach-code.rpgle:10:3: warning: code will never be executed [-Wunreachable-code]
   return 'fail here';
   ^
